### PR TITLE
[15.0][MIG] sale_order_archive: Migration to 15.0

### PR DESCRIPTION
**Pull request by CRogos**
Follow up of #1916 
rebased and unit tests added

### CODE REVIEW

**Comment by CRogos** [comment]
@dhx , @ajaniszewska-dev you have reviewed #1916
I have rebased it and added some unit tests because of failing check tasks. Could you review?

**Review by dhx — APPROVED** [review]

**Review by AnilKesariya — APPROVED** [review]

**Review by AnilKesariya — APPROVED** [review]

**Review by AnilKesariya — APPROVED** [review]

**Review by AnilKesariya — APPROVED** [review]

**Review by AnilKesariya — APPROVED** [review]

**Review by AnilKesariya — APPROVED** [review]

File: sale_order_archive/tests/test_sale_order_archive.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    def test_sale_order_archive_toggle_active(self):
+        self.assertTrue(self.sale_order.active)
+        self.sale_order.state = "done"
+        self.sale_order.toggle_active()
+        self.assertFalse(self.sale_order.active)
+
+    def test_sale_order_archive_donot_toggle_active(self):
+        self.assertTrue(self.sale_order.active)
+        with self.assertRaises(UserError):
```
Comment (mmequignon) [inline sale_order_archive/tests/test_sale_order_archive.py]: please use [assertRaisesRegex](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex) to ensure that the exception that is raised is yours
```suggestion
        regex = r"Only 'Locked' or 'Canceled' orders can be archived"
        with self.assertRaisesRegex(UserError, regex):
```

**Review by mmequignon — APPROVED** [review]

**Comment by CRogos** [comment]
@rousseldenis  maybe close the other PR first and retry?

**Comment by rousseldenis** [comment]
Merging this one as no answer on the other side.

/ocabot merge patch

**Pull request merged**